app/ide-desktop/lib/dashboard/src/services/LocalBackend.ts

```diff
@@ -489,4 +720,47 @@ export default class LocalBackend extends Backend {
   override getCheckoutSession() {
     return this.invalidOperation()
   }
+
+  /** Run the Project Manager with the given command-line arguments. */
+  private async runProjectManagerCommand<T = void>(
+    body: BodyInit | null,
+    name: string,
+    ...cliArguments: string[]
+  ): Promise<T> {
+    const searchParams = new URLSearchParams({
+      // The names come from a third-party API and cannot be changed.
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      'cli-arguments': JSON.stringify([`--${name}`, ...cliArguments]),
+    }).toString()
+    const response = await fetch(
+      `${this.baseUrl}/api/run-project-manager-command?${searchParams}`,
+      {
+        method: 'POST',
+        body,
+      }
+    )
+    // There is no way to avoid this as `JSON.parse` returns `any`.
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-argument
+    const json: projectManager.JSONRPCResponse<never> = await response.json()
+    if ('result' in json) {
+      return json.result
+    } else {
+      throw new Error(json.error.message)
+    }
+  }
+
+  /** List directories, projects and files in the given folder. */
+  private async projectManagerFilesystemList(parentId: backend.DirectoryId | null) {
+    /** The type of the response body of this endpoint. */
+    interface ResponseBody {
+      readonly entries: FileSystemEntry[]
+    }
+
+    const response = await this.runProjectManagerCommand<ResponseBody>(
+      null,
+      'filesystem-list',
+      parentId ?? this.defaultRootDirectory
+    )
+    return response.entries
+  }
```

Comment:
How come some projectManager-interacting methods are defined as private functions here, and some are defined like `projectManager.deleteAsset`. Seems like we should be all-in on one or the other, no?

Reply:
one kind uses the existing WS connection, the other kind (this kind) goes through a HTTP proxy defined by Electron to `exec` the actual project manager binary - supposedly these endpoints are not exposed in WS for security purposes (the user invoking the command needs to have access to the files to be able to `exec` the binary)

Reply:
i suppose they should be defined on the project manager class regardless...